lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +255,94 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
-        pass
+        article_body = article_soup.find('div', {'class': 'article__body'})
+        article_text = article_body.find_all(['p', 'div'], {'class': 'quote-text'})
+        art_text = [i.text for i in article_text]
+        self.article.text = '\n'.join(art_text)
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        title = article_soup.find('h1', {'class': 'title'}).text
+        if title:
+            self.article.title = title
+        author = article_soup.find('span', {'class': 'toolbar-opposite__author-text'}).text
+        if author:
+            self.article.author.append(author)
+        date_bs = article_soup.find('time', {'class': 'toolbar__text'}).get('datetime')
+        if isinstance(date_bs, str):
+            date_and_time = re.search(r'\d{4}-\d{2}-\d{2}', date_bs).group() + \
+                        ' ' + re.search(r'\d{2}:\d{2}:\d{2}', date_bs).group()
+            self.article.date = self.unify_date_format(date_and_time)
+        topic = article_soup.find('a', {'class': 'toolbar__item toolbar__main-link'}).text
+        if topic:
+            self.article.topics.append(topic)
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        return datetime.datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        soup = make_request(self.full_url, self.config)
+        article_bs = BeautifulSoup(soup.text, 'lxml')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive crawler implementation
+    """
+
+    def __init__(self, config: Config) -> None:
+        """
+        Initializes the instance of CrawlerRecursive class
+        """
+        super().__init__(config)
+        self.start_url = config.get_seed_urls()[0]
+
 
 
 def main() -> None:
     """
     Entrypoint for scrapper module
     """
-    pass
+    configuration = Config(path_to_config=CRAWLER_CONFIG_PATH)
+    prepare_environment(ASSETS_PATH)
+    crawler = Crawler(configuration)
+    crawler.find_articles()
+    print(crawler.get_search_urls())
+    print(len(crawler.get_search_urls()))
+    for i, full_url in enumerate(crawler.urls, start=1):
+        parser = HTMLParser(full_url=full_url, article_id=i, config=configuration)
+        article = parser.parse()
+        if isinstance(article, Article):
```

Comment:
do you in fact just that it is not None?

Reply:
In case i do not check if the article value belongs to Article class, i have mistakes in mypy, and this is the only way i've found to pass the checks